-1 
Tested at: 2d38fa717103659ed3cf78697d7c61e7847fb1ec
 I found an error when building: 
 
 <pre>>> Compiling edm plugin /build/cmsbuild/jenkins-workarea/workspace/ib-any-integration/CMSSW_7_4_X_2015-01-23-1400/src/GeneratorInterface/GenFilters/src/PythiaFilterTTBar.cc 
>> Compiling edm plugin /build/cmsbuild/jenkins-workarea/workspace/ib-any-integration/CMSSW_7_4_X_2015-01-23-1400/src/GeneratorInterface/GenFilters/src/PythiaFilterZJet.cc 
>> Compiling edm plugin /build/cmsbuild/jenkins-workarea/workspace/ib-any-integration/CMSSW_7_4_X_2015-01-23-1400/src/GeneratorInterface/GenFilters/src/PythiaFilterZJetWithOutBg.cc 
>> Compiling edm plugin /build/cmsbuild/jenkins-workarea/workspace/ib-any-integration/CMSSW_7_4_X_2015-01-23-1400/src/GeneratorInterface/GenFilters/src/PythiaFilterZgamma.cc 
/build/cmsbuild/jenkins-workarea/workspace/ib-any-integration/CMSSW_7_4_X_2015-01-23-1400/src/GeneratorInterface/GenFilters/src/PythiaFilterGammaGamma.cc: In member function 'virtual bool PythiaFilterGammaGamma::filter(edm::Event&, const edm::EventSetup&)':
/build/cmsbuild/jenkins-workarea/workspace/ib-any-integration/CMSSW_7_4_X_2015-01-23-1400/src/GeneratorInterface/GenFilters/src/PythiaFilterGammaGamma.cc:167:8: error: 'this_id' was not declared in this scope
        this_id = (*itSeed)->pdg_id();
        ^
/build/cmsbuild/jenkins-workarea/workspace/ib-any-integration/CMSSW_7_4_X_2015-01-23-1400/src/GeneratorInterface/GenFilters/src/PythiaFilterGammaGamma.cc:168:8: error: 'mom' was not declared in this scope
        mom = (*itSeed);
        ^
</pre> 
 you can see the results of the tests here: 
 https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-6976/2077/summary.html

--- a/GeneratorInterface/GenFilters/src/PythiaFilterGammaGamma.cc
+++ b/GeneratorInterface/GenFilters/src/PythiaFilterGammaGamma.cc
@@ -124,6 +124,10 @@ bool PythiaFilterGammaGamma::filter(edm::Event& iEvent, const edm::EventSetup& i
   std::vector<TLorentzVector> candidate, candidateNarrow, candidateSeed;
   std::vector<const GenParticle*>::iterator itSeed;
 
+  const GenParticle* mom;
+  int this_id;
+  int first_different_id;
+
   for(itSeed = seeds.begin(); itSeed != seeds.end(); ++itSeed) {
 
     TLorentzVector energy, narrowCone, temp1, temp2, tempseed;

